lib/movie_like_button.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:flutter/material.dart';
+
+class MovieLikeButton extends StatefulWidget {
+  const MovieLikeButton({super.key});
+
+  @override
+  State<MovieLikeButton> createState() => _MovieLikeButtonState();
+}
+
+class _MovieLikeButtonState extends State<MovieLikeButton> {
+  int likesAmount = 2341;
+  IconData likeIcon = Icons.favorite_border;
+
+  void toggleLike() {
+    if (likeIcon == Icons.favorite_border) {
+      likesAmount++;
+      likeIcon = Icons.favorite;
+    }
+    else if (likeIcon == Icons.favorite) {
+      likesAmount--;
+      likeIcon = Icons.favorite_border;
+    }
+
+    setState(() {});
+  }
```

Comment:
Instead to compare by type of icon you can create a local variable like 'isLikeTapped' with to posibble values (true/false).
When you display the icon on the ElevatedButton, you can do
icon: isLikeTapped ? Icon(Icons.favorite_border) : Icon(Icons.favorite_border)